Show message when comments are turned off

syeopite opened the pull request:
Fixes: #1092
I want to be awarded the bounty associated to the issue this PR is fixing.

![comments](https://github.com/iv-org/invidious/assets/70992037/73a83540-2fc4-491b-959d-0b0a317833bb)

I've also added a message for when comments are disabled via Invidious config.

syeopite committed:
Add error message when comments are disabled

syeopite committed:
Add btn to try reddit comments when yt is disabled

syeopite committed:
Escape warning messages for disabled comments

syeopite committed:
Add missing semicolons

Fijxu commented:
Tested and checked, works without problems. I haven't tested it on production but it works as is, no need to change settings. However i wasn't able to find the invidious config related to "added a message for when comments are disabled via Invidious config" but it works. Comments on videos WITHOUT comments disabled are displayed.

If anyone wants to test this in a debug/test instance it can be easily merged using this commands:

```sh
$ git fetch origin pull/4051/head:show-comments-turned-off-message
$ git merge show-comments-turned-off-message
$ make
```

![image](https://github.com/iv-org/invidious/assets/61166695/d04e9d03-8fc1-4585-ac2e-25aa5f868557)

*I would add translations for `es.json` in this pull request but I don't know how to propose changes to a pull request*

syeopite commented:
> However i wasn't able to find the invidious config related to "added a message for when

Oops. I shouldn't have used the word config there. I meant when comments are disabled in Invidious's preferences.  

See https://github.com/iv-org/invidious/issues/2585#issuecomment-1686633019 for how to do so. Its not very clear in the UI. 

> _I would add translations for es.json in this pull request but I don't know how to propose changes to a pull request_

Unfortunately Github only lets you suggest changes on lines near changes that occurred within the PR. If you'd like to add translations feel free to make a PR to the `show-comments-turned-off-message` branch on my fork here https://github.com/syeopite/invidious. It should get automatically included in this PR once merged.

syeopite committed:
Update comment detection to not use msg renderer

SamantazFox commented on assets/js/watch.js:
```diff
@@ -219,7 +219,27 @@ function get_youtube_comments() {
+            <div id="comments-turned-off-on-video-message" class="h-box v-box">
+                <p><b>${video_data.youtube_comments_disabled_text}</b></p>
+
+                <p>
+                    <b> 
+                        <a href="javascript:void(0)" data-comments="reddit" id="try-reddit-comments-link"> 
+                            ${video_data.youtube_comments_disabled_try_reddit}
+                        </a> 
+                    </b>
+                </p> 
+            </div>`;
```
```suggestion
                <p><b><a href="javascript:void(0)" data-comments="reddit" id="try-reddit-comments-link">
                    ${video_data.youtube_comments_disabled_try_reddit}
                </a></b></p>
            </div>`;
```

SamantazFox replied:
This can be condensed a bit ^^
Also, I'm not sure about `javacsript:void(0)` links (See #769)

syeopite committed:
Use short syntax for comment entry-point detection

Co-authored-by: Samantaz Fox <coding@samantaz.fr>

syeopite committed:
Improve locale keys for comments disabled messages

Co-authored-by: Samantaz Fox <coding@samantaz.fr>

syeopite committed:
Remove extra whitespace

syeopite committed:
Condense try-reddit-comments-link HTML in player.js

Co-authored-by: Samantaz Fox <coding@samantaz.fr>

syeopite committed:
Use button instead of anchor for try reddit link

syeopite committed:
Request Reddit comm when yt disabled during nojs=1

syeopite committed:
Change order of commentsEnabled in parse results

syeopite committed:
Lint

syeopite committed:
Rebase error

Anjolao commented:
The Backend on invidious is not working. Is there a way to fix the backends?

Fijxu commented:
Merged into my instance as it works fine

![image](https://github.com/user-attachments/assets/e55a0088-b6cd-469b-a2fc-b892ab6cdfab)

unixfox reviewed (commented):
Good for merging

Fijxu commented on src/invidious/videos/parser.cr:
```diff
@@ -416,6 +416,18 @@ def parse_video_info(video_id : String, player_response : Hash(String, JSON::Any
+
+  # When comments are enabled there should be a comments-entry-point section in the primary results
+  if primary_results
+    section = primary_results.as_a.find(&.dig?("itemSectionRenderer", "sectionIdentifier").== "comments-entry-point")
```
`comments-entry-point` has changed to `comment-item-section`

```suggestion
    section = primary_results.as_a.find(&.dig?("itemSectionRenderer", "sectionIdentifier").== "comment-item-section")
```

unixfox reviewed (approved):
Good after resolving the merge conflicts.

Fijxu committed:
update the naming of the comments section in Innertube

Downstreamed from https://git.nadeko.net/Fijxu/invidious/commit/36ac67610d1e52df731a21525fac966413541fa9

Fijxu commented on src/invidious/views/watch.ecr:
```diff
@@ -289,14 +298,41 @@ we're going to need to do it here in order to allow for translations.
+                                <%= comment_html %>
+                            <% else %>
+                                <a href="/watch?<%= env.params.query %>&nojs=1">
+                                    <%= I18n.translate(locale, "comments_youtube_disabled_try_reddit_no_js") %>
```
Missing HTML.escape call here. I'll add a commit to `master` branch for this.
